test/testdata/infer/generics/arity.rb

```diff
@@ -0,0 +1,40 @@
+# typed: true
+extend T::Sig
```

Comment:
Can you add a test for the cases where `NotABox` and `Box` are passed the current amount of generic args?

Reply:
:portal-from: DM: these tests are already covered by the rest of Sorbet's test suite and we're going to opt to not add more redundant tests here.